pkg/reconciler/revision/resources/queue_test.go

```diff
@@ -401,202 +359,162 @@ func TestMakeQueueContainerWithPercentageAnnotation(t *testing.T) {
-				PodSpec: corev1.PodSpec{
-					Containers: []corev1.Container{{
-						Name:           containerName,
-						ReadinessProbe: testProbe,
-						Resources: corev1.ResourceRequirements{
-							Requests: corev1.ResourceList{
-								corev1.ResourceName("cpu"):    resource.MustParse("50m"),
-								corev1.ResourceName("memory"): resource.MustParse("128Mi"),
-							},
+		rev: revision(
+			withContainerConcurrency(1),
+			func(revision *v1.Revision) {
+				revision.Spec.TimeoutSeconds = ptr.Int64(45)
+				revision.ObjectMeta = metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "bar",
+					UID:       "1234",
+					Labels: map[string]string{
+						serving.ServiceLabelKey: "svc",
+					},
+					Annotations: map[string]string{
+						serving.QueueSideCarResourcePercentageAnnotation: "0.2",
+					},
+				}
+				revision.Spec.PodSpec.Containers = []corev1.Container{{
+					Name:           containerName,
+					ReadinessProbe: testProbe,
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceName("cpu"):    resource.MustParse("50m"),
+							corev1.ResourceName("memory"): resource.MustParse("128Mi"),
 						},
-					}},
-				},
-			},
-		},
-		want: &corev1.Container{
-			// These are effectively constant
-			Name: QueueContainerName,
-			Resources: corev1.ResourceRequirements{
-				Requests: corev1.ResourceList{
+					},
+				}}
+			}),
+		want: queueContainer(
+			func(c *corev1.Container) {
+				c.Env = env(map[string]string{
+					"SERVING_SERVICE": "svc",
+				})
+				c.Resources.Requests = corev1.ResourceList{
 					corev1.ResourceName("cpu"):    resource.MustParse("25m"),
 					corev1.ResourceName("memory"): resource.MustParse("50Mi"),
-				},
-			},
-			Ports:           append(queueNonServingPorts, queueHTTPPort),
-			ReadinessProbe:  defaultKnativeQReadinessProbe,
-			SecurityContext: queueSecurityContext,
-			// These changed based on the Revision and configs passed in.
-			Image: "alpine",
-			Env: env(map[string]string{
-				"SERVING_SERVICE": "svc",
+				}
+				c.Image = "alpine"
 			}),
-		}}, {
+	}, {
 		name: "Invalid resources percentage in annotations",
-		rev: &v1.Revision{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: "foo",
-				Name:      "bar",
-				UID:       "1234",
-				Labels: map[string]string{
-					serving.ServiceLabelKey: "svc",
-				},
-				Annotations: map[string]string{
-					serving.QueueSideCarResourcePercentageAnnotation: "foo",
-				},
-			},
-			Spec: v1.RevisionSpec{
-				ContainerConcurrency: ptr.Int64(1),
-				TimeoutSeconds:       ptr.Int64(45),
-				PodSpec: corev1.PodSpec{
-					Containers: []corev1.Container{{
-						Name:           containerName,
-						ReadinessProbe: testProbe,
-						Resources: corev1.ResourceRequirements{
-							Requests: corev1.ResourceList{
-								corev1.ResourceName("cpu"):    resource.MustParse("50m"),
-								corev1.ResourceName("memory"): resource.MustParse("128Mi"),
-							},
+		rev: revision(
```

Comment:
I'd think `revision` would take "name" and "ns" as params (or hardcoded, given it's UT).
This should really simplify the metadata updates.